contracts/ExodiaAccessControlUpgradeable.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+pragma abicoder v2;
+
+import "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import "./ExodiaRoles.sol";
+
+/**
+ * The ExodiaAccessControlUpgradeable let's contract implement role based 
+ * permission using the ExodiaRoles contract.
+ * It adds the Machine role on a per contract basis.
+ * The Machine role is intended to be used if the contract can only be called from another
+ * contract or by a keeper we know the address.
+ */
+abstract contract ExodiaAccessControlUpgradeable is Initializable {
```

Comment:
I would rename this contract ExodiaAccessControlInitializable as it inherits Initializable